content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```diff
@@ -693,6 +699,78 @@ kubectl apply -f my-crontab.yaml
 crontab "my-new-cron-object" created
 ```
 
+## Validation Rules
+
+{{< feature-state state="alpha" for_k8s_version="v1.23" >}}
+
+Validation Rules are in alpha since 1.23 and validate custom resources when the
+`CustomResourceValidationExpressions` [feature
+gate](/docs/reference/command-line-tools-reference/feature-gates/) enabled and the schema is a
+[structural schema](#specifying-a-structural-schema).
+
+Validation rules use the [CEL](https://github.com/google/cel-spec) expression language to validate custom
+resource values. Validation rules are included in CustomResourceDefinition schemas using the
+`x-kubernetes-validation-rules` extension.
+
+For example:
+
+```yaml
+    ...
+    openAPIV3Schema:
+      type: object
+      properties:
+        spec:
+          type: object
+            x-kubernetes-validation-rules:
+              - rule: "self.minReplicas <= self.replicas && self.replicas <= self.maxReplicas"
```

Comment:
For teaching, would this work better as two separate rules in the array?

Reply:
Nifty, yes. Updated.